packages/runner/src/builder/module.ts

```diff
@@ -225,3 +226,27 @@ export function derive<In, Out>(...args: any[]): OpaqueRef<any> {
 // unsafe closures: like derive, but doesn't need any arguments
 export const computed: <T>(fn: () => T) => OpaqueRef<T> = <T>(fn: () => T) =>
   lift<any, T>(fn)(undefined);
+
+/**
+ * action: Creates a handler that doesn't use the event parameter.
+ *
+ * This is to handler as computed is to lift/derive:
+ * - User writes: action(() => count.set(count.get() + 1))
+ * - Transformer rewrites to: handler((_, { count }) => count.set(count.get() + 1))({ count })
+ *
+ * The transformer extracts closures and makes them explicit, just like how
+ * computed(() => expr) becomes derive({}, () => expr) with closure extraction.
+ *
+ * @param fn - A function that receives captured state and performs side effects
+ * @returns A HandlerFactory that can be bound to state
+ */
+export function action<T>(
+  fn: (props: T) => void,
+): HandlerFactory<T, void>;
+export function action<T>(
+  fn: (props: T) => void,
+): HandlerFactory<T, void> {
+  // Wrap the action callback to match handler's (event, props) signature
+  // The { proxy: true } option tells handler to accept schema-less callbacks
+  return handler<void, T>((_, props) => fn(props), { proxy: true });
```

Comment:
oh, that's wrong. but then i don't think this should ever be called directly, as the transformed will rewrite that to `handler`, no?

maybe throw an error here?

(the non-CTS variant that's correct would be `return handler<T, void>((event) => fn(event))` but this is almost never going to be useful since there is nothing `fn` could write to -- so better just throw and remind user to add the CTS enable marker)